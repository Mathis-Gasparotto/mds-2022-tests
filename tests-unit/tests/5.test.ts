import { beforeEach, describe, expect, test } from 'vitest'
import { Picture, User, UserStore, useUserStore } from '../5'

const user: User = {
  id: 1,
  name: 'Maggio'
}

const pictures: Picture[] = [
  {
    id: 1,
    userId: 1,
    url: 'oui.com'
  }
]

describe('userStore', () => {
  beforeEach(() => {
    const userStore: UserStore = useUserStore.getState()
    userStore.reset()
  })

  test('setCurrentUser', async () => {
    let userStore: UserStore = useUserStore.getState()
    await userStore.setCurrentUser(user)

    userStore = useUserStore.getState()

    expect(userStore.currentUser).toMatchSnapshot()
  })

  test('addPicture', async () => {
    let userStore: UserStore = useUserStore.getState()
    await userStore.addPicture(pictures[0])

    userStore = useUserStore.getState()

    expect(userStore.pictures).toMatchSnapshot()
  })

  test('deletePicture', async () => {
    let userStore: UserStore = useUserStore.getState()
    await userStore.addPicture(pictures[0])
    await userStore.deletePicture(1)

    userStore = useUserStore.getState()

    expect(userStore.pictures).toMatchSnapshot()
  })
})
