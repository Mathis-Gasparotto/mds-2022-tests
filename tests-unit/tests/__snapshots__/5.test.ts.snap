// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`userStore > addPicture 1`] = `
[
  {
    "id": 1,
    "url": "oui.com",
    "userId": 1,
  },
]
`;

exports[`userStore > deletePicture 1`] = `[]`;

exports[`userStore > setCurrentUser 1`] = `
{
  "id": 1,
  "name": "Maggio",
}
`;
